lib/tail/index.ts

```diff
@@ -1,3 +1,7 @@
 import { AnyArray } from "../any-array";
 
-export type Tail<Type extends AnyArray> = Type extends [any, ...infer Rest] ? Rest : never;
+export type Tail<Type extends AnyArray> = Type extends readonly []
+  ? never
+  : Type extends readonly [any?, ...infer Rest]
```

Comment:
QQ: is it important that we use `any?` and not `any`?

I wouldn't see the difference if you asked me, because `any` is a top type, so from my understanding, leaving `[any, ...]` should work fine as well

If it doesn't, my bad, happy to accept it as is

Reply:
> QQ: is it important that we use any? and not simply any?

Yes, the optional modifier is actually necessary, it doesn't work without that.

<br>

Consider the example of `[string?]` and `[any]`. The tuple `[string?]` doesn't extend `[any]` because `[any]` defines a tuple with exactly one element, however, `[string?]` allows for the possibility of an empty array.

```ts
type Test1 = [string?] extends [any] ? true : false; // Without optional
//   ^? type Test1 = false

type Test2 = [string?] extends [any?] ? true : false; // With optional
//   ^? type Test2 = true
```

This is similar to how an optional property wouldn't extend a non-optional `any` property:
```ts
type Test1 = { a?: string } extends { a: any } ? true : false; // Without optional
//   ^? type Test1 = false

type Test2 = { a?: string } extends { a?: any } ? true : false; // With optional
//   ^? type Test2 = true
```

Playground: https://tsplay.dev/WPQrYw

Reply:
Great, it's clear now! Thanks